### PR TITLE
[SPARK-49410][SQL][TESTS] Update collation benchmarks

### PR DESCRIPTION
**Pull request by uros-db**
### What changes were proposed in this pull request?
Updating collation benchmarks to compute relative time, instead of relative speed. Also, re-run all collation benchmarks (after introducing various perf improvements as part of the recent collation effort).

### Why are the changes needed?
Relative speed is displayed in format: "x0.0" for collations that are slower than the referent collation (UTF8_BINARY).

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existing tests.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Please regenerate and attach the result, @uros-db .

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/test/scala/org/apache/spark/benchmark/Benchmark.scala [hunk 1]
```diff
@@ -94,9 +94,11 @@ private[spark] class Benchmark(
   /**
    * Runs the benchmark and outputs the results to stdout. This should be copied and added as
    * a comment with the benchmark. Although the results vary from machine to machine, it should
-   * provide some baseline.
+   * provide some baseline. If `relative_time` is set to `true`, the `Relative` column will be
+   * the relative time of each case relative to the first case (less is better). Otherwise, it
+   * will be the relative execution speed of each case relative to the first case (more is better).
    */
-  def run(): Unit = {
+  def run(relative_time: Boolean = false): Unit = {
```
[the remark this answers is not in the document]
Comment (uros-db) [reply core/src/test/scala/org/apache/spark/benchmark/Benchmark.scala]: fixed

**Comment by MaxGekk** [comment]
@uros-db Please, regenerate benchmark results for Java 17/21, see:
- https://github.com/apache/spark/pull/32015
- https://github.com/apache/spark/pull/47743

File: core/src/test/scala/org/apache/spark/benchmark/Benchmark.scala [hunk 2]
```diff
@@ -116,14 +118,15 @@ private[spark] class Benchmark(
       name + ":", "Best Time(ms)", "Avg Time(ms)", "Stdev(ms)", "Rate(M/s)", "Per Row(ns)", "Relative")
```
Comment (MaxGekk) [inline core/src/test/scala/org/apache/spark/benchmark/Benchmark.scala:118]: I believe the benchmark result should be self-readable. A reader shouldn't consult to source code to figure out meaning of `Relative`. How about to indicate that in the header.

**Review by MaxGekk — CHANGES_REQUESTED** [review]
@uros-db Please, regenerate benchmark results after changing the headers.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @uros-db and @dongjoon-hyun for review.

**Pull request closed without merging**